### DISCUSSION TITLE
Refill printer paper info message

### DISCUSSION THREAD

**Discussion by maehw** [xtech/self-o-mat#27] [Ideas]
Hi!

I and others are probably using a relatively low-cost/consumer grade Canon Selphy CP 1200/1300 printer.
The photo paper try is quite small and needs to be refilled regularly.

Is there a way to introduce a "refill printer paper" info message for maintenance?
Not every guest/participant of an event might be familiar or confident to do it themself - so this could be useful to call somebody for support and not to wonder why there's no printout (which might look obvious).

Is there a way to get info about missing paper from CUPS and forward it to the user?

**Comment by ClemensElflein** [comment]
Hi!

the software should already show a warning if the paper is empty.
You can then use the software normally, but a permanent warning should be shown on the live view.

As soon as the printer paper is refilled, the warning should disappear and printing should resume as normal.

**Reply by maehw** [reply]
Alright, then it might depend on my printer. I just reconstructed the scenario.

CUPS tells that the printer is still processing ("waiting for job to complete"). But on the printer display it's telling me that it has detected that either the cassette is not inserted properly (which is) or that it has run out of paper. After inserting new paper it prints and then goes back to idle state.

The model I am using is a Canon SELPHY CP-1200. Driver is drvless.ppd, version 1.28.7. But that's probably rather related to the firmware, I guess. No clue how I can find out a version or update it.

Have you also tried this with a Canon SELPHY?

**Reply by ClemensElflein** [reply]
Yes, Canon Selphy printers worked fine. Note that the message pops up as soon as the first page fails to print. So if you print and then the paper is empty, there will not yet a message be shown. As soon as another print attempt is made, the message should show.

**Reply by ClemensElflein** [reply]
I think it is a driver issue. For our firmware image, we used the "selphy print" CUPS driver. It needs to be compiled and installed manually, though. Unfortunately, I don't have a guide here.

After some research, I found the repository:
https://git.shaftnet.org/cgit/selphy_print.git

If you get it to work, it would be huge if you could write the commands to build and install this to the wiki. If I remember correctly, it's just clean and install, but I'm not sure..
